modules/options/src/main/scala/scala/build/actionable/ActionablePreprocessor.scala

```diff
@@ -0,0 +1,27 @@
+package scala.build.actionable
+
+import scala.build.Logger
+import scala.build.options.BuildOptions
+
+case object ActionablePreprocessor {
+  val actionableHandlers = Seq(
+    ActionableDependencyHandler
+  )
+
+  def generateActionableDiagnostics(
+    options: BuildOptions,
+    logger: Logger
+  ): Seq[ActionableDiagnostic] =
+    actionableHandlers.flatMap { handler =>
+      val values = handler.extractPositionedOptions(options)
+      val (errors, actionableDiagnostic) =
+        values.map(v => handler.createActionableDiagnostic(v, options)).partitionMap(identity)
+      errors.map(logger.debug(_))
+      actionableDiagnostic.flatten
+    }
```

Comment:
Not sure we should discard errors this way here… Better accumulate them and keep them. Something like this ought to do:
```suggestion
  ): Either[BuildException, Seq[ActionableDiagnostic]] =
    actionableHandlers
      .map { handler =>
        handler
          .extractPositionedOptions(options)
          .map(v => handler.createActionableDiagnostic(v, options))
          .sequence
          .left.map(CompositeBuildException(_))
      }
      .sequence
      .left.map(CompositeBuildException(_))
      .map(_.flatten)
```

Reply:
This might need `import scala.build.Ops._`

The errors should then be propagated up to `scala.cli.commands.DependencyUpdate`, where we can call `.orExit(logger)`, or `.orReport(logger)` (in watch mode).